tests/integration_test/run_integration_test.py

```diff
@@ -0,0 +1,142 @@
+# Util imports
+from input_output.file_output import load_graph_list_from_filepath, create_csv_file, write_csv_line
+from input_output.sys_output import fail, passed
+import tkinter as tk
+from tkinter import filedialog
+from time import time
+# Algorithm imports
+from tests.integration_test.settings import *
+from tests.integration_test.solutions import *
+from tests.integration_test.isomorphism_problem import are_isomorph, amount_of_isomorphisms
+
+"""
+General integration test for the Graph Isomorphisms problem.
+Settings of the algorithm must be changed in > settings.py <
+"""
+
+"""
+FILE INPUT
+"""
+# Select files to evaluate
+root = tk.Tk()
+root.withdraw()
+file_paths = filedialog.askopenfilenames()
+
+"""
+FILE OUTPUT
+"""
+# If csv must be created, create an empty file here
+if write_to_csv:
+    csv_filepath = create_csv_file("isomorphism-problem-test")
+    # Write first row with column names
+    csv_column_names = ['file', 'graph1', 'graph2',
+                        'are_isomorph expected result', 'are_isomorph actual result', 'passed', 'are_isomorph processing time (s)',
+                        'amount_of_isomorphisms expected result', 'amount_of_isomorphisms actual result', 'passed', 'amount_of_isomorphisms processing time (s)']
+    write_csv_line(csv_filepath, csv_column_names)
+
+"""
+ACTUAL TEST
+"""
+# For every file that needs to be evaluated, all combinations of graphs within that file are evaluated.
+graph_count = 0
+error_count = 0
+for filepath in file_paths:
+    graphs = load_graph_list_from_filepath(filepath)
+    filename = (filepath.split("/")[-1]).split(".")[0]
+    solution_map = {}
+    if filename in solution:
+        solution_map = solution[filename]
+
+    for i in range(0, len(graphs) - 1):
+        for j in range(i + 1, len(graphs)):
+            G = graphs[i]
+            H = graphs[j]
+
+            if console_pass or console_fail:
+                print('---------------------------')
+                print("Statistics of " + filename + "-" + str(i) + "_" + str(j) + ":")
+                print('---------------------------')
+
+            start_isomorph = time()
+            are_isomorph_actual = are_isomorph(G.copy(), H.copy())
+            end_isomorph = time()
+
+            start_amount_isomorphisms = time()
+            amount_isomorph_actual = amount_of_isomorphisms(G, H)
+            end_amount_isomorphisms = time()
+
+            # If solution of file is in solutions.py, check if solution is correct
+            if bool(solution_map):
+                graph_count += 1
```

Comment:
Volgens mij tel je nu het aantal grafen waar geen oplossing van bekend is niet mee. Is dat de bedoeling?

Reply:
Dat was wel de bedoeling, omdat dat uiteindelijk laten zien wordt als het aantal grafen dat getest wordt. Als je maar 1 graaf test, maar er daarnaast nog 10 andere runt die geen solution hebben, dan lijkt het anders alsof je 11 grafen test. En als alles dan slaagt lijkt het super goed. Terwijl als je weet dat het maar 1 is, je misschien even moet denken of je nog anderen moet testen.

Reply:
Maar is het dan uiteindelijk ook niet een resultaat als je erachter komt dat twee grafen niet isomorf zijn? Misschien een idee om een aparte counter bij te houden voor echt alle grafen en eentje voor de isomorfe grafenparen?

Reply:
Maar als je toch niet weet of dat antwoord klopt of niet, waarom zou je daar dan een counter voor willen hebben?
Ik had het zo bedacht omdat we met dat toernooitje misschien ook deze wel willen aanroepen, maar we weten natuurlijk niet wat het juiste antwoord is. Dus dat we die niet meetellen in de graph_count (die ik eventueel test_graph_count kan noemen). En ik zou een counter kunnen maken voor het totaal aantal grafen dat we vergeleken hebben?